recipes-sota/auto-provisioning/auto-provisioning/auto-provisioning.sh

```diff
@@ -157,4 +168,30 @@ function main()
     log "Device successfully provisioned"
 }
 
+# Source the overrides script.
+#
+# Currently the script is expected to expose two functions, namely:
+#
+# - get_device_id: if defined, it should output the "device ID" to be used when
+#   registering the device to the platform.
+# - get_device_name: uf defined, it should output the "device name" to be used
```

Comment:
Small typo here "uf defined"

Reply:
Thanks for the review.

Regarding the error handling, I think all we can do is to document this behavior so a user can add to their overrides script whatever fallback logic they need to ensure a name/ID is always produced even in case the main method fails.

Also, notice that the behavior here is just the same as with the other various errors that cause the provisioning script to fail today, most of which are unlikely to fix themselves. If that was an issue for some user, they could review the service configuration to limit the number of retries. If they did so, a device would require a reboot or a manual restart of the service in order to try again which the user would need to be okay with.